### PR TITLE
New profile: mariadbd_akonadi

### PR DESCRIPTION
**Pull request by curiosityseeker**
A similar approach is available on https://invent.kde.org/pim/akonadi/-/tree/master/apparmor?ref_type=heads where also @cboltz contributed.

This profile has some additional rules and is formatted according to the guidelines of this project.

### CODE REVIEW

File: apparmor.d/groups/kde/mariadbd_akonadi [hunk 1]
```diff
@@ -0,0 +1,54 @@
+abi <abi/3.0>,
+
+#include <tunables/global>
+
+@{exec_path} = @{bin}/mariadb{,-*} @{bin}/mariadbd @{bin}/mysql @{bin}/mysqld{,*}
+profile mariadbd_akonadi @{exec_path} flags=(complain) {
+  include <abstractions/base>
+  include <abstractions/bash>
+  include <abstractions/consoles>
+  include <abstractions/nameservice>
```
Comment (roddhjav) [inline apparmor.d/groups/kde/mariadbd_akonadi:10]: Use nameservice-strict. And add network access in the profile.

File: apparmor.d/groups/kde/mariadbd_akonadi [hunk 2]
```diff
@@ -0,0 +1,54 @@
+abi <abi/3.0>,
+
+#include <tunables/global>
+
+@{exec_path} = @{bin}/mariadb{,-*} @{bin}/mariadbd @{bin}/mysql @{bin}/mysqld{,*}
+profile mariadbd_akonadi @{exec_path} flags=(complain) {
+  include <abstractions/base>
+  include <abstractions/bash>
```
Comment (roddhjav) [inline apparmor.d/groups/kde/mariadbd_akonadi:8]: The bash abstraction is only needed for interactive shell. So not here.

**Comment by roddhjav** [comment]
What don't you simply use the profiles they provide?

File: apparmor.d/groups/kde/mariadbd_akonadi [hunk 3]
```diff
@@ -0,0 +1,54 @@
+abi <abi/3.0>,
+
+#include <tunables/global>
+
+@{exec_path} = @{bin}/mariadb{,-*} @{bin}/mariadbd @{bin}/mysql @{bin}/mysqld{,*}
```
Comment (roddhjav) [inline apparmor.d/groups/kde/mariadbd_akonadi:5]: That won't work. Doing so you are going to confine a mariadbd server. Not the database of akonadi.

File: apparmor.d/groups/kde/mariadbd_akonadi (hunk 2 above)
Comment (curiosityseeker) [reply apparmor.d/groups/kde/mariadbd_akonadi:8]: That abstraction was in the original profile. I have to check the consequences once removed.

File: apparmor.d/groups/kde/mariadbd_akonadi (hunk 3 above)
Comment (curiosityseeker) [reply apparmor.d/groups/kde/mariadbd_akonadi:5]: I'm not sure that I understand. In my understanding AppArmor profiles do not confine data(bases) (unless specific data is protected by including,e.g., the private-files or sensitive-home abstractions or adding specific deny rules) but rather applications are confined by allowing them to do/access only what's necessary. So how is this profile (and the original one) different from all the other ones? 

FWIW, I've been using this profile for quite some time in enforce mode:
`/usr/bin/mariadbd (8323) mariadbd_akonadi`

I'll remove the bash abs and replace the nameservice abs with nameservice-strict  and see what's necessary to add.

File: apparmor.d/groups/kde/mariadbd_akonadi (hunk 1 above)
Comment (curiosityseeker) [reply apparmor.d/groups/kde/mariadbd_akonadi:10]: That was in the original profile.

**Comment by curiosityseeker** [comment]
Is there still something missing? If you don't want to accept this PR I can delete it, no problem.

**Comment by roddhjav** [comment]
Sorry for the delay. I would rather focus on profile that need to be done than on profile that are already packaged (plus they would be some change to be done to actually merge it).

**Pull request closed without merging**